src/java.base/share/classes/java/lang/runtime/SwitchBootstraps.java

```diff
@@ -769,11 +778,12 @@ private static boolean unconditionalExactnessCheck(Class<?> selectorType, Class<
             return true;
         }
         else if (selectorType.equals(targetType) ||
-                ((selectorType.equals(byte.class) && !targetType.equals(char.class)) ||
-                 (selectorType.equals(short.class) && (selectorWrapper.isStrictSubRangeOf(targetWrapper))) ||
-                 (selectorType.equals(char.class)  && (selectorWrapper.isStrictSubRangeOf(targetWrapper)))  ||
-                 (selectorType.equals(int.class)   && (targetType.equals(double.class) || targetType.equals(long.class))) ||
-                 (selectorType.equals(float.class) && (selectorWrapper.isStrictSubRangeOf(targetWrapper))))) return true;
+                (targetType.isPrimitive() && selectorType.isPrimitive() &&
+                    ((selectorType.equals(byte.class) && !targetType.equals(char.class)) ||
```

Comment:
Will `unconditionalExactnessMatch(byte.class, boolean.class` return `true`? I think it shouldn't, even if `isNotValidPair` is called before.

Reply:
`unconditionalExactnessMatch` needs to return an answer whether a pair is unconditional assuming it is applicable. Fusing them would need to elaborate the type of the returned answer.